membership-attribute-service/app/services/AttributesFromZuora.scala

```diff
@@ -215,5 +221,40 @@ object AttributesFromZuora {
 
     regular.map(_.copy(DigitalSubscriptionExpiryDate = maybeLatestDate)).orElse(gift)
   }
+
+  val metrics = Metrics("AttributesFromZuora") //referenced in CloudFormation
+
+  def alertIfAttributesDoNotMatch(fromZuora: Option[Attributes], fromSupporterProductData: Option[Attributes]): Unit =
+    if (attributesDoNotMatch(fromZuora, fromSupporterProductData)) {
+      logger.warn(
+        "There was a mismatch between the attributes returned by Zuora and the supporter-product-data data store\n" +
+        s"Zuora returned: $fromZuora\n" +
+        s"supporter-product-data returned: $fromSupporterProductData"
+      )
+      metrics.put("AttributesMismatch", 1) //referenced in CloudFormation
+    } else logger.info("attributes returned from Zuora match those returned from supporter-product-data")
+
+  def attributesDoNotMatch(fromZuora: Option[Attributes], fromSupporterProductData: Option[Attributes]) =
+    !(
+      fromZuora.isEmpty && fromSupporterProductData.isEmpty ||
+        (fromZuora.map(_.Tier) == fromSupporterProductData.map(_.Tier) &&
+          fromZuora.map(_.RecurringContributionPaymentPlan) == fromSupporterProductData.map(_.RecurringContributionPaymentPlan) &&
+          datesAreEqualEnough(fromZuora.flatMap(_.DigitalSubscriptionExpiryDate), fromSupporterProductData.flatMap(_.DigitalSubscriptionExpiryDate)) &&
+          datesAreEqualEnough(fromZuora.flatMap(_.PaperSubscriptionExpiryDate), fromSupporterProductData.flatMap(_.PaperSubscriptionExpiryDate)) &&
+          datesAreEqualEnough(fromZuora.flatMap(_.GuardianWeeklySubscriptionExpiryDate), fromSupporterProductData.flatMap(_.GuardianWeeklySubscriptionExpiryDate))
+          )
+      )
+
+  def datesAreEqualEnough(fromZuora: Option[LocalDate], fromSupporterProductData: Option[LocalDate]) = {
+    //Currently dates from the supporter-product-data store will be a day after the zuora term end date to avoid
+    // worrying about time zones and when new subscriptions are created. This may well change.
+    (fromZuora.isEmpty && fromSupporterProductData.isEmpty) || (for{
+        zuora <- fromZuora
+        supporter <- fromSupporterProductData
+      } yield {
+      zuora == supporter || zuora.plusDays(1) == supporter
+    }).getOrElse(false)
```

Comment:
I think it's the same as this
```suggestion
    fromZuora == fromSupporterProductData || 
      fromZuora.map(_.plusDays(1)) == fromSupporterProductData
```